src/main/kotlin/nmcp/NmcpAggregationExtension.kt

```diff
@@ -0,0 +1,71 @@
+package nmcp
+
+import gratatouille.GExtension
+import nmcp.internal.configureAttributes
+import nmcp.internal.nmcpConsumerConfigurationName
+import nmcp.internal.task.registerPublishTask
+import org.gradle.api.Action
+import org.gradle.api.Project
+import org.gradle.api.tasks.bundling.Zip
+
+@GExtension(pluginId = "com.gradleup.nmcp.aggregation")
+open class NmcpAggregationExtension(private val project: Project) {
+    internal val spec = project.objects.newInstance(NmcpSpec::class.java)
+
+    internal val consumerConfiguration = project.configurations.create(nmcpConsumerConfigurationName) {
+        it.isCanBeResolved = true
+        it.isCanBeConsumed = false
+
+        it.configureAttributes(project)
+    }
+
+    val zipTaskProvider = project.tasks.register("zipAggregation", Zip::class.java) {
+        it.archiveFileName.set("aggregation.zip")
+        it.destinationDirectory.set(project.layout.buildDirectory.dir("nmcp/zip"))
+        it.from(consumerConfiguration.elements.map {
+            check (it.isNotEmpty()) {
+                "nmcp: aggregation is empty. Make sure to create publications in sub-projects and specify projects to publish by adding them to the 'nmcpAggregation' configuration."
+            }
+            it.map {
+                project.zipTree(it)
+            }
+        })
+    }
+
+    init {
+        project.registerPublishTask(
+            taskName = "publishAggregationToCentralPortal",
+            inputFile = zipTaskProvider.flatMap { it.archiveFile },
+            spec = spec
+        )
+    }
+
+    /**
+     * Configures the central portal parameters
+     */
+    fun centralPortal(action: Action<NmcpSpec>) {
```

Comment:
It might be worth consider renaming `NmcpSpec` to something that makes sense for the end users. For instance, `CentralPortalConfiguration` or `CentralPortalExtension`, etc. `NmcpSpec` would be visible in the build script APIs and it looks like a technical name.

Reply:
Agreed. I went with `CentralPortalOptions` as both `Configuration` and `Extension` are very loaded terms in Gradle-land.